spec/index.html

```diff
@@ -1802,42 +1814,59 @@ <h3>RDF Triples Constructors</h3>
         |curPredicate| <var>N</var> .</span>
     </p>
 
-    <h4 id="tripleTerm" style="padding-bottom:0; margin-bottom:0;"><span>Triple Terms:</span></h4>
+    <h4 id="tripleTerm" style="padding-bottom:0; margin-bottom:0;"><span>Triple Terms</span></h4>
 
     <p>Beginning the <a href="#grammar-production-tripleTerm"><code>tripleTerm</code></a> production
       records the |curSubject| and |curPredicate|.
       Finishing the <a href="#grammar-production-tripleTerm"><code>tripleTerm</code></a> production
-      and restores the recorded values of |curSubject| and |curPredicate|.</p>
+      restores the recorded values of |curSubject| and |curPredicate|.</p>
+
+    <h4 id="reifier" style="padding-bottom:0; margin-bottom:0;"><span>Reifier</span></h4>
+
+    <p>Beginning the <a href="#grammar-production-reifier"><code>reifier</code></a> production,
+      if |curReifier| is set, the production yields the RDF triple |curReifier| <code>rdf:reifies</code> |curTripleTerm|.
```

Comment:
> if |curReifier| is set,

It is not set until the end of the production so it is that previous sate at this point.

The production will set `curReifier` - there 's no "if". Do this now and not at the end, then yield the reifies triple using `curReifier`.

```suggestion
    |curReifier| is taken from the <a href="#handle-reifier"><code>reifier</code></a> term constructor.
    For the |tripleTerm| from the last RDF triple yielded then yield the the RDF triple |curReifier| <code>rdf:reifies</code> |curTripleTerm|.
```
There is no need to have a stack for `curReifier`. Even in annotations, only the last value is needed.

Reply:
The text is intended to capture the events generated by `:s :p :o ~:r1 ~:r2`. When seeing `~:r1` there is no previously established `curReifier` so nothing is emitted, then `curReifier` is set from the `reifier` term constructor. The next time when it sees `~:r2`, `curReifier` would be `:r1` so `:ri rdf:reifies <<(...)>>` is emitted and then `curReifier` is set to `:r2`.

There is no stack for `curReifier`, there is just if it was previously set or not (Note that there is no text that says to save the value of `curReifier` anywhere). It is cleared when entering Annotations`, set in either `Reifier` or `Annotation Block`, and consumed in `Reifier`, `Annotation Block` or `Annotations`.

Reply:
I'm beginning to see what you want to do here but then some text does seem right.

> Finishing the [annotation](https://pr-preview.s3.amazonaws.com/w3c/rdf-turtle/pull/62.html#grammar-production-annotation) production yields the RDF triple curReifier rdf:reifies curTripleTerm, resets the value of curReifier

needs a condition that curReifier is set. In `:s :p :o ~:r {| :x :y |}` it is not (it's been used). That's where I see identical triples being emitted once for the  annotationBlock and one at the and of annotations.

There needs to be more recording and restoring:

```
  :s :p :o ~:r1 {| :x1 :y2 ~:r2 {| :xx1 :yy2 |} ;
                   :x2 :y2
                 |}.
```
which I think is:
```
:s :p :o .
:r1 rdf:reifies <<( :s :p :o )>> .
:r1 :x1 :y2 .
:r2 rdf:reifies <<( :r1 :x1 :y2 )>> .
:r2 :xx1 :yy2 .
:r1 :x2 :y2 .
```
note `:r1 :x2 :y2`.

Reply:
That's what my parser produces.